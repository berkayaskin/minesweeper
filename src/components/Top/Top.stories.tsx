import { Meta, Story } from '@storybook/react/types-6-0'
import type { TopComponentType } from './Top'
import { Top } from './Top'

export default {
  title: 'Top/Top',
  component: Top,
} as Meta

const Template: Story<TopComponentType> = (args) => <Top {...args} />

export const TopPanel = Template.bind({})
TopPanel.args = {
  title: 'Minesweeper',
  feature: 'Flag',
  firstAction: 'Ctrl',
  secondAction: 'Click',
}
