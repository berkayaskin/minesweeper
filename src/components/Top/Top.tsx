import { Container } from '@chakra-ui/react'
import { Header, HeaderProps } from './Header'
import { Legend, LegendProps } from './Legend'

export type TopComponentType = HeaderProps & LegendProps

export const Top = ({ title, ...legendProps }: TopComponentType) => {
  return (
    <Container maxWidth='sm'>
      <Header title={title} />
      <Legend {...legendProps} />
    </Container>
  )
}
