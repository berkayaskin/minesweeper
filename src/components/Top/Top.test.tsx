import { render } from '@testing-library/react'
import { Top } from './Top'

it('should render top correctly', () => {
  const { asFragment } = render(
    <Top feature='Flag' firstAction='ctrl' secondAction='click' title='Minesweeper' />,
  )
  expect(asFragment()).toMatchSnapshot()
})
