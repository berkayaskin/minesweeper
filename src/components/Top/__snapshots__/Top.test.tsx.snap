// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render top correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-1ms27t5"
  >
    <h2
      class="chakra-heading css-6gnk2m"
    >
      Minesweeper
    </h2>
    <code
      class="css-g5hezf"
    >
      <div
        class="chakra-stack css-14lxv93"
      >
        <h1
          class="chakra-text css-1adld2q"
        >
          Flag:
        </h1>
        <span
          class="chakra-text css-91rsr4"
        >
          ctrl
        </span>
        <p
          class="chakra-text css-1x3w0s1"
        >
          +
        </p>
        <span
          class="chakra-text css-1bjr0e8"
        >
          click
        </span>
      </div>
    </code>
  </div>
</DocumentFragment>
`;
